File: graph-builder/sample-repo/typescript/sample.ts
/**
 * Sample TypeScript file for testing
 */

/**
 * User interface
 */
export interface User {
  id: string;
  name: string;
  email: string;
}

/**
 * User service class
 */
export class UserService {
  private users: Map<string, User> = new Map();

  /**
   * Get user by ID
   * @param id User ID
   * @returns User object or undefined
   */
  getUserById(id: string): User | undefined {
    return this.users.get(id);
  }

  /**
   * Create a new user
   * @param user User object
   */
  createUser(user: User): void {
    this.users.set(user.id, user);
  }

  /**
   * Validate user email
   * @param email Email address
   * @returns True if valid
   */
  validateEmail(email: string): boolean {
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    return emailRegex.test(email);
  }
}

/**
 * Fetch user from API
 * @param userId User ID
 * @returns Promise with user data
 */
export async function fetchUserFromAPI(userId: string): Promise<User> {
  const response = await fetch(`/api/users/${userId}`);
  const data = await response.json();
  return data as User;
}

/**
 * Calculate user age
 * @param birthYear Birth year
 * @returns Current age
 */
export function calculateAge(birthYear: number): number {
  const currentYear = new Date().getFullYear();
  return currentYear - birthYear;
}

File: graph-builder/sample-repo/python/sample.py
"""
Sample Python file for testing
"""

from typing import Optional, Dict
import re


class User:
    """User class"""
    
    def __init__(self, user_id: str, name: str, email: str):
        """Initialize user"""
        self.id = user_id
        self.name = name
        self.email = email


class UserService:
    """User service class"""
    
    def __init__(self):
        """Initialize user service"""
        self.users: Dict[str, User] = {}
    
    def get_user_by_id(self, user_id: str) -> Optional[User]:
        """Get user by ID"""
        return self.users.get(user_id)
    
    def create_user(self, user: User) -> None:
        """Create a new user"""
        self.users[user.id] = user
    
    def validate_email(self, email: str) -> bool:
        """Validate user email"""
        email_regex = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
        return bool(re.match(email_regex, email))


async def fetch_user_from_api(user_id: str) -> User:
    """Fetch user from API"""
    # Simulated API call
    import aiohttp
    async with aiohttp.ClientSession() as session:
        async with session.get(f'/api/users/{user_id}') as response:
            data = await response.json()
            return User(**data)


def calculate_age(birth_year: int) -> int:
    """Calculate user age"""
    from datetime import datetime
    current_year = datetime.now().year
    return current_year - birth_year

